### ISSUE TITLE
"No native build was found" during `yarn install --verbose` is disconcerting

### ISSUE THREAD

**Issue by jrose-signal** [mafintosh/node-gyp-build#42]
`node-gyp-build`-the-executable currently detects whether there's a build or prebuild available by running a script that loads `node-gyp-build`-the-library, which (reasonably) throws an error if the build/prebuild is missing. It also only shows stderr output from this script in "verbose" mode, which is also reasonable. However, this means that `yarn install --verbose` of a library using `node-gyp-build` can end up with an "error" in the log output, even though it's just how `node-gyp-build` is detecting whether there's a build or not. This can lead to confusion, especially if there's an actual problem later on (see signalapp/libsignal#456).

I personally think the best solution is to separate the error-throwing from the search. This would allow other errors to still be displayed, as well as distinguishing between "the build is missing" and "the build is present but cannot be loaded" (rebuilding may not help in the latter case).

**Comment by mafintosh** [comment]
What's your suggestion for updating this? Not fully sure I follow. If users use the `node-gyp-build` cli it will fallback to classic npm behaivour of building the source if no prebuild works - you can disable this with the `--optional` flag. Are you saying we should never do that?

**Comment by jrose-signal** [comment]
No, no! I'm saying it's worth taking extra steps to avoid *printing* the error, even in `--verbose` mode. The actual behavior is good.

**Comment by vweevers** [comment]
I've found it useful many times. Would it help to tweak the message?
